### PR TITLE
feat: Inspect checluster status and fail if status.reason is not empty

### PR DESCRIPTION
**Pull request by tolusha**
Signed-off-by: Anatolii Bazko <abazko@redhat.com>

### What does this PR do?
- Inspect checluster status and fail if status.reason is not empty (when there is an error on operator side)

### Screenshot/screencast of this PR

```
    ❯ Operator pod bootstrap
      ✖ Scheduling...failed
        → Eclipse Che operator failed, reason: Fail, message: Fail for some reason. Consider increasing error recheck timeout with --k8spoderrorrechecktimeout flag.
        Downloading images
        Starting
      Prepare Eclipse Che cluster CR
 ›   Error: Error: Eclipse Che operator failed, reason: Fail, message: Fail for some reason. Consider increasing error recheck timeout with --k8spoderrorrechecktimeout flag.
 ›   Command server:deploy failed. Error log: /home/tolusha/.cache/chectl/error.log
 ```

### What issues does this PR fix or reference?

https://github.com/eclipse/che/issues/19284

### How to test this PR?

1. Deploy Eclipse Che using custom operator image with setting not empty status details
```
deploy.SetStatusDetails(deployContext, "Fail", "Fail for some reason", "")
```

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [ ] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [ ] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

File: src/tasks/kube.ts [hunk 1]
```diff
@@ -92,12 +106,21 @@ export class KubeTasks {
           const taskTitle = task.title
           const iterations = this.kubeHelper.podReadyTimeout / this.interval
           for (let i = 1; i <= iterations; i++) {
-            const failedState = await this.getFailedWaitingState(namespace, selector, 'Running')
-            if (failedState) {
+            // check cheCluster status
+            const cheClusterFailState = await this.getCheClusterFailState(namespace)
+
+            const failedState = await this.getContainerFailState(namespace, selector, 'Running')
+            if (cheClusterFailState || failedState) {
```
Comment (mmorhun) [inline src/tasks/kube.ts:113]: I would suggest to fail only on operator failure and silently proceed to further check/recheck.

File: src/tasks/kube.ts [hunk 2]
```diff
@@ -28,18 +32,28 @@ export class KubeTasks {
...
           for (let i = 1; i <= iterations; i++) {
+            // check cheCluster status
+            const cheClusterFailState = await this.getCheClusterFailState(namespace)
+
             // check 'PodScheduled' condition
-            const failedCondition = await this.getFailedPodCondition(namespace, selector, 'PodScheduled')
-            if (failedCondition) {
+            const podFailState = await this.getPodFailState(namespace, selector, 'PodScheduled')
+
+            if (cheClusterFailState || podFailState) {
```
[the remark this answers is not in the document]
Comment (tolusha) [reply src/tasks/kube.ts:41]: agree

**Comment by tolusha** [comment]
@mmorhun 
Addressed you remarks

**Review by mmorhun — APPROVED** [review]

**Comment by l0rd** [comment]
@benoitf I think that the milestone has failed to be assigned here

**Pull request merged**